source/MaterialXCore/Element.h

```diff
@@ -71,6 +71,52 @@ using ElementMap = std::unordered_map<string, ElementPtr>;
 /// A standard function taking an ElementPtr and returning a boolean.
 using ElementPredicate = std::function<bool(ConstElementPtr)>;
 
+/// @class ElemenEquivalenceOptions
+/// A set of options for controlling for equivalence comparison.
+class MX_CORE_API ElementEquivalenceOptions
+{
+  public:
+    ElementEquivalenceOptions() 
+    {
+        format = Value::getFloatFormat();
+        precision = Value::getFloatPrecision();
+        skipAttributes = {};
+        ignoreAttributeOrder = true;
+        skipValueComparisons = false;
+        status = EMPTY_STRING;
+    };
+    ~ElementEquivalenceOptions() { }
+
+    /// Floating point format option for floating point value comparisons
+    Value::FloatFormat format;
+
+    /// Floating point precision option for floating point value comparisons
+    int precision;
+
+    /// Attribute filtering options. By default all attributes are considered.
+    /// Name, category attributes cannot be skipped.
+    /// 
+    /// For example UI attribute comparision be skipped by setting:
+    /// skipAttributes = { 
+    ///     ValueElement::UI_MIN_ATTRIBUTE, ValueElement::UI_MAX_ATTRIBUTE,
+    ///     ValueElement::UI_SOFT_MIN_ATTRIBUTE, ValueElement::UI_SOFT_MAX_ATTRIBUTE,
+    ///     ValueElement::UI_STEP_ATTRIBUTE, Element::XPOS_ATTRIBUTE, 
+    ///     Element::YPOS_ATTRIBUTE };
+    StringSet skipAttributes;
+
+    /// Option to indicate whether to ignore the order that attributes
+    /// are specified on an element. Default is to ignore order.
+    bool ignoreAttributeOrder;
+
+    /// Do not perform any value comparisions. Instead perform exact string comparisons for attributes
+    /// Default is false. The operator==() method can be used instead as it always performs
+    /// a strict comparison. Default is false.
+    bool skipValueComparisons;
+
+    /// Status string for first difference found.
+    string status;
```

Comment:
I really like having `ElementEquivalenceOptions`, its going to allow us to expand on this functionality much easier in the future.

I'm not sure `status` really feels like it belongs in the equivalence options object. I can imagine creating a single version of this object and wanting to re-use it for many comparisons. 

Perhaps instead we could have a `ElementEquivalenceResult` object that is returned.  Which stores both the `true/false` success of the check, but optionally the status as well.  I can imagine cases where perhaps we might want the equivalence to run as fast as possible and not be constructing strings, if we don't care about why.  I can also imagine cases where we might not want the equivalence to exist with `false` at the first difference, but instead run over the entire document and report as many differences are is possible, then we might want a `vector<string>`. This `ElementEquivalenceResult` could have a bool operator defined, so it could still be treated like the `true/false` result that it is most likely to be used for. 

Alternately we could keep the boolean return value and just carry the status result as a separate argument.

Reply:
I think it's more consistent and convenient to return value a `boolean`. Then like the `validate(string *message)` method status can be returned if the argument is specified. 

Instead of a string it can be a actual struct to allow for future changes without a signature. `ElementEquivalenceResult` sounds good to me. It can contain a list of "differences". For now it will return the first difference.

Reply:
I like that, it would be nice to also have the ability to not require the explanation at all - as an optimization if its just a computer doing an equivalence check and a human doesn't need to know why.

Reply:
See if what I've stuck in looks good or not. Only if the optional argument is specified it will log info otherwise it does nothing.